components/net_traits/bluetooth_scanfilter.rs

```diff
@@ -0,0 +1,132 @@
+
+const MAX_DEVICE_NAME_LENGTH: usize = 29;
+
+#[derive(Deserialize, Serialize)]
+pub struct ServiceUUIDSequence(Vec<String>);
+
+impl ServiceUUIDSequence {
+
+    pub fn new(vec: Vec<String>) -> ServiceUUIDSequence {
+        ServiceUUIDSequence(vec)
+    }
+
+    fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct BluetoothScanfilter {
+    name: String,
+    name_prefix: String,
+    services: ServiceUUIDSequence,
+}
+
+impl BluetoothScanfilter {
+    fn is_empty_or_invalid_filter(&self) -> bool {
+        self.name.is_empty() && self.name_prefix.is_empty() &&
+        self.services.is_empty() &&
+        self.name.len() > MAX_DEVICE_NAME_LENGTH &&
+        self.name_prefix.len() > MAX_DEVICE_NAME_LENGTH
+    }
+
+    pub fn new(name: String, name_prefix: String, services: Vec<String>) -> BluetoothScanfilter {
+        BluetoothScanfilter {
+            name: name,
+            name_prefix: name_prefix,
+            services: ServiceUUIDSequence::new(services),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct BluetoothScanfilterSequence(Vec<BluetoothScanfilter>);
+
+impl BluetoothScanfilterSequence {
+    pub fn new(vec: Vec<BluetoothScanfilter>) -> BluetoothScanfilterSequence {
+        BluetoothScanfilterSequence(vec)
+    }
+}
+
+impl BluetoothScanfilterSequence {
+    fn has_empty_or_invalid_filter(&self) -> bool {
+        self.0.is_empty() &&
+        self.0.iter().all(|x| !(x.is_empty_or_invalid_filter()))
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct RequestDeviceoptions {
+    // _filters: BluetoothScanfilterSequence,
+    filter: BluetoothScanfilter,
+    optional_services: ServiceUUIDSequence,
+}
+
+impl RequestDeviceoptions {
+    pub fn new(// filters: BluetoothScanfilterSequence,
+               filter: BluetoothScanfilter,
+               services: ServiceUUIDSequence)
+        -> RequestDeviceoptions {
+            RequestDeviceoptions {
+                // _filters: filters,
+                filter: filter,
+                optional_services: services,
+            }
+    }
+    //TODO replace this with method get_filters after sequnce dictionaries are avialable(zakorgy)
+    pub fn get_filter(&self) -> &BluetoothScanfilter {
+        &self.filter
+    }
+
+    /*pub fn get_filters(&self) -> &BluetoothScanfilterSequence {
+        &self.filters
+    }*/
+}
+
+pub fn matches_filter(device: &BluetoothDevice, filter: &BluetoothScanfilter) -> bool {
+    if filter.is_empty_or_invalid_filter() {
+        return false;
+    }
+    // TODO: replace unwrap(zakorgy)
+    let device_name: String = device.get_name().unwrap();
+
+    if !filter.name.is_empty() && (device_name != filter.name) {
+        return false;
+    }
+
+    if !filter.name_prefix.is_empty() && (!device_name.starts_with(&*filter.name_prefix)) {
+        return false;
+    }
+
+    if !filter.services.is_empty() {
+        // TODO: replace unwrap(zakorgy)
```

Comment:
no; there should be a new line between functions, including their comments, but this comment is fine here